index/slashing/slashing.go

```diff
@@ -9,30 +9,37 @@ import (
 	"github.com/ipfs/go-cid"
```

Comment:
Similar as others:
- refactored/fixes
- save/load index from datastore
- extracted types to `types.go`